### ISSUE TITLE
Minor issue with generating PAR::Packed package

### ISSUE THREAD

**Issue by kiwiroy** [jhthorsen/applify#8]
## brief

When using [PAR::Packer](https://metacpan.org/pod/pp) and `documentation __FILE__;` in an Applify script `--help` fails at this [eval](https://github.com/jhthorsen/applify/blob/df2b22ba4ea09a34e4745332ebbe4775496887b4/lib/Applify.pm#L363).

The issue is demonstated in c94ac9e and an interim fix is present in 1004a89, which although [passing tests](https://travis-ci.org/kiwiroy/applify/builds/400659428), might not be the best solution.

## actual behaviour

```bash
$ git co c94ac9e
$ ./example/par-pack.sh ./example/test1.pl
/tmp/RvqKLXTJGq syntax OK
Could not load script/test1.pl: Search pattern not terminated at (eval 15) line 1.
```

## expected behaviour

```bash
$ ./example/par-pack.sh ./example/test2.pl 
/tmp/vAWiQ6bGgT syntax OK
Usage:
   --iiiiiii     asd
    -x           asd
   --input-file  File to read from
   --output-dir  Directory to write files to
 * --dry-run     Use --no-dry-run to actually do something

   --help        Print this help text
   --man         Display manual for this application
   --version     Print application name and version
```

## reference

Only an issue when`__FILE__` is used.

```patch
--- example/test1.pl 2018-07-06 09:59:46.923737759 +1200
+++ example/test2.pl 2018-07-06 10:38:14.492562430 +1200
@@ -9,7 +9,7 @@
 option flag => dry_run => 'Use --no-dry-run to actually do something', required => 1;
 
 version 1.23;
-documentation __FILE__;
+documentation $0;
 
 sub app::generate_exit_value {
   return int rand 100;

```

## tl;dr

Use `$0` instead of `__FILE__` as parameter to `documentation` if using `pp`.

**Comment by jhthorsen** [comment]
I'm not sure if this a good idea, since $0 can be changed. I'm also not sure how this would behave on Windows.

Mentioning it as a potential workaround for this case is something I don't mind though.

**Comment by kiwiroy** [comment]
Any mention, should also refer to [PAR::Packer docs](https://github.com/rschupp/PAR-Packer/blob/master/contrib/docs/who_am_i.txt).
